Ardupilog.m

```diff
@@ -83,52 +84,136 @@ function delete(obj)
             else
                 warn('File not closed successfully')
             end
-
-            % The number of header occurances is a good estimate at the number of messages
-            % -- (If the header occurs "accidentally" in the data, this count will be wrong)
-            obj.numMsgs = length(strfind(obj.log_data, obj.header));
             
-            % Prompt user to choose how many messages to process, and setup GUI waitbar
-            num_lines = input(sprintf('How many log lines to process? (%d total, press Enter to process all): ', obj.numMsgs));
-            waitbar_max_lines = num_lines;
-            if isempty(num_lines)
-                num_lines = obj.numMsgs*10; % numMsgs should be accurate, the *10 is "just in case"...
-                waitbar_max_lines = obj.numMsgs;
-            end
-            obj.wb_handle = waitbar(0, 'Initializing...', ...
-                         'Name', ['Processing log: ', obj.fileName], ...
-                         'CreateCancelBtn', 'setappdata(gcbf, ''cancel_pressed'', 1)');
-            setappdata(obj.wb_handle,'cancel_pressed',0);
-
-            % Process messages one by one, either creating formats, moving to seen, or appending seen
-            for ctr = 1:num_lines % HGM: Consider replacing for-loop with while loop?
-                % If another log line exists, process it
-                if obj.log_data_read_ndx < numel(obj.log_data);
-                    % Check to see if user pressed cancel button
-                    if getappdata(obj.wb_handle, 'cancel_pressed') == 1
-                        disp('Canceled by user')
-                        break
-                    end
-                    % The main call to process a single log line
-                    obj.readLogLine();
-                else % at end of file
-                    disp('Reached end of log.')
-                    obj.numMsgs = obj.lastLineNum;
-                    break
+            % Discover the locations of all the messages
+            FMTLength = 89;
+            FMTIndices = obj.discoverMSG(128,FMTLength);
+            % Read the FMT message
+            % Generate the index mask which corresponds to the places where
+            % FMT information exists
+            indices = reshape(ones(length(FMTIndices),1)*(3:(FMTLength-1)) + (ones(length(FMTIndices),1).*FMTIndices')*ones(1,FMTLength-3),[1 length(FMTIndices)*(FMTLength-3)]);
```

Comment:
This line is hard for me to parse and understand. If it's okay with you, I might break it into a few lines or add more comments in a future commit.